File: leetcode2363.py
class Solution:
    def mergeSimilarItems(self, items1: List[List[int]], items2: List[List[int]]) -> List[List[int]]:
      
        d: dict = {}
        for k, v in items1:
          d[k] = d.get(k,0) + v
        for k, v in items2:
          d[k] = d.get(k,0) + v
        
        result = []
        
        for k, v in d.items():
          result.append([k,v])
        result.sort()
        return result

File: leetcode2363.cpp
//bruteforce algorithm
class Solution {
public:
  
  
  vector<vector<int>> mergeSimilarItems(vector<vector<int>>& items1, vector<vector<int>>& items2) {
    for(int i = 0;i<items2.size();i++){
      bool k = true;
      for(int j = 0;j<items1.size();j++){
        if(items2[i][0] == items1[j][0]){
          items1[j][1] += items2[i][1];
          k = true;
          break;
        }else{
          k = false;
        }
      }
      if(!k){
        items1.push_back(items2[i]);
      }
    }
    sort(begin(items1),end(items1));
    return items1;
  }
};
